src/odemis/acq/align/test/autofocus_test.py

```diff
@@ -602,5 +603,76 @@ def test_multi_det(self):
         self.assertEqual(len(res), ngs * nds)
 
 
+class TestAutofocus1d(unittest.TestCase):
+    """
+    Test autofocus functions on 1 line CCD.
+    """
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            test.start_backend(SPARC2_FOCUS_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # find components by their role
+        cls.ccd = model.getComponent(role="ccd")
+        cls.spectrometer = model.getComponent(role="spectrometer-integrated")
+
+        cls.focus = model.getComponent(role="focus")
+        cls._good_focus = cls.focus.position.value["z"]
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        test.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+    @timeout(1000)
+    def test_autofocus_spect(self):
+        """
+        Test AutoFocus on 1 line CCD for example spectrum.
+        """
+        self.focus.moveAbs({"z": self._good_focus - 200e-6}).result()
+        self.ccd.updateMetadata({model.MD_LENS_RI: 1.0e-2})
+        f = align.AutoFocus(self.spectrometer, None, self.focus, method=MTD_BINARY)
+        foc_pos, foc_lev = f.result(timeout=900)
+        logging.debug("Found focus at {} good focus at {}".format(foc_pos, self._good_focus))
+        numpy.testing.assert_allclose(foc_pos, self._good_focus, atol=2.5e-5)
```

Comment:
I guess this 2.5e-5 is not coming completely out of nowhere, but just big enough to work?
As the focus step size is set (in the microscope file) to 10.9e-6m, that's essentially 2 focus steps. That sounds fine, but could be documented.

In test_autofocus_slit(), that sounds a little bit too much optimistic, as it'll fail if it's not exactly the right step.